python/plugins/per-project-location/location/per_project_location.py

```diff
@@ -0,0 +1,85 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2016-2022 ftrack
+
+
+import os
+import sys
+import platform
+import functools
+import tempfile
+import logging
+
+import ftrack_api
+from ftrack_api.entity.location import Location
+from ftrack_api.accessor.disk import DiskAccessor
+from ftrack_api.structure.standard import StandardStructure
+
+
+logger = logging.getLogger('per_project_location')
+
+
+class PerProjectLocation(Location):
+    platform = platform.system()
+
+    def _get_project_root(self, component):
+
+        target_project_id = component['version']['project_id']
+
+        # use :storage / project location: from project settings
+        target_project = self.session.query(
+            f'select disk, root from Project where id is "{target_project_id}"'
+        ).first()
+
+        root_folder = target_project['root']
+        if not root_folder:
+            root_folder = target_project['disk'].get(
```

Comment:
The disk section here doesn't work on macOS. `platform.system().lower()` returns `darwin` for me, which is not in `['windows', 'unix', 'id', 'projects', 'name']` that `target_project['disk']` consists of, which in turn results in that macOS tries to publish to `/unix` (read-only path in the root of the OS).

@henriknorin-ftrack must have been trying this on Windows, or only tried with a root folder set? (not the disks option)